main.ts

```diff
@@ -1,142 +1,140 @@
-
-
 /**
  *  UTILS  AND TEMPLATES
  */
 
-
- // Images for Simple slider
- const images: string[] = Array(7)
- .fill(1)
- .map((i, index) => `${index + 1}.jpeg`); // [1.jpeg, 2.jpeg ...]
-
+// Images for Simple slider
+const images: string[] = Array(7)
+  .fill(1)
+  .map((i, index) => `${index + 1}.jpeg`); // [1.jpeg, 2.jpeg ...]
 
 // Simple Slider Template
 const simpleSliderTemplate = (source: string) => `
    <img src="${source}" alt="" />
 `;
 
 const createDOMElFromString = (domstring: string): HTMLElement => {
- const html = new DOMParser().parseFromString(domstring, "text/html");
- return <HTMLElement>html.body.firstChild;
+  const html = new DOMParser().parseFromString(domstring, "text/html");
+  return <HTMLElement>html.body.firstChild;
 };
 
 // Slider Classes
 abstract class BaseSlider {
- protected imagePath = "./assets/";
-
- public currentIndex = 0;
- public el: HTMLElement;
- protected slides: HTMLElement[] = [];
-
- // TODO: add union type for slides
- abstract addSlides(slides: string[]): void;
-
- constructor(selector: string, private showButtons: boolean = true) {
-   this.el = document.querySelector(selector);
- }
-
- // TODO: next and previous methods need to be fixed
- // when slides index > slides.length you can see empty slide
-
- public next(): void {
-   this.showSlide(++this.currentIndex);
- }
-
- public previous(): void {
-   this.showSlide(--this.currentIndex);
- }
-
- protected showSlide(index: number): void {
-   this.slides.forEach((item, i) => {
-     if (index === i) {
-       item.classList.remove("is-hidden");
-       return;
-     }
-
-     item.classList.add("is-hidden");
-   });
- }
-
- private addButtons(): void {
-   const buttonsContainer = document.createElement("div");
-   buttonsContainer.classList.add("navigation");
-   buttonsContainer.classList.add("has-text-centered");
-   buttonsContainer.innerHTML = `
+  protected imagePath = "./assets/";
+
+  public currentIndex = 0;
+  public el: HTMLElement;
+  protected slides: HTMLElement[] = [];
+
+  // TODO: add union type for slides
+  abstract addSlides(slides: string[] | ITextSlide[]): void;
+
+  constructor(selector: string, private showButtons: boolean = true) {
+    this.el = document.querySelector(selector);
+  }
+
+  private nextIndex(distance: number): number {
```

Comment:
I would like to move this method from BaseSlider and make it part of the number type. As I understane I should use prototypes for this but I'm not sure about it. So what I'm trying to get it something with next interface
```
this.currentIndex.next(this.slides.lenghth)
this.currentIndex.previous(this.slides.lenghth)
```
or 
```
this.currentIndex = this.currentIndex.next(this.slides.lenghth, 1)
this.currentIndex = this.currentIndex.next(this.slides.lenghth, -1)
```
How I can achive that, thx

Reply:
haven't fully got the idea

but, could be this way with getters/setters

get currentIndex() {
  return {
   next(...)  {
  },
 previous(...) {
 }
 }
}